app/controllers/tag_controller.rb

```diff
@@ -359,7 +359,7 @@ def delete
     node_tag = NodeTag.where(nid: params[:nid], tid: params[:tid]).first
     node = Node.where(nid: params[:nid]).first
     # only admins, mods, and tag authors can delete other peoples' tags
-    if node_tag.uid == current_user.uid || logged_in_as(['admin', 'moderator']) || (node.uid == current_user.uid && node_tag.name != "locked")
+    if (node_tag.uid == current_user.uid && !node.has_tag('locked')) || logged_in_as(['admin', 'moderator']) || (node.uid == current_user.uid && !node.has_tag('locked'))
```

Comment:
This is with reference to [this](https://github.com/publiclab/plots2/pull/9709#discussion_r644721788) in PR #9709.

Reply:
Thanks for adding this @Manasa2850, this looks good to me. I am curious about the comment in line 361...

`# only admins, mods, and tag authors can delete other peoples' tags` 
from the code, it seems that even a node_tag author or the author of the node cannot delete a tag as long as a node contains the `locked` tag?

So maybe the comment should be `# only admins, mods can delete other peoples' tags if the tags contain the locked tag`